src/arch/riscv/interrupts.cc

```diff
@@ -0,0 +1,229 @@
+                mask.uei = mask.uti = mask.usi = 1;
+                /* local interrupts are enabled globally if delegated
+                 * to lower privilege mode and bit uie is 1 */
+                mask.local = mask.local | sideleg.local;
+            }
+            break;
+        case PRV_S:
+            /* local interrupts are enabled globally if not delgated
+             * to lower privilege mode */
+            mask.local = ~mideleg.local;
+            mask.mei = (!mideleg.mei) | (mideleg.mei & status.sie);
+            mask.mti = (!mideleg.mti) | (mideleg.mti & status.sie);
+            mask.msi = (!mideleg.msi) | (mideleg.msi & status.sie);
+            if (status.sie) {
+                mask.sei = mask.sti = mask.ssi = 1;
+                /* local interrupts are enabled globally if delegated
+                 * to lower privilege mode and bit sie is 1 */
+                mask.local = mask.local | mideleg.local;
+            }
+            mask.uei = mask.uti = mask.usi = 0;
+            break;
+        case PRV_M:
+
+            if (status.mie) {
+                    mask.local = gem5::mask(48);
+                    mask.mei = mask.mti = mask.msi = 1;
+            }
+            mask.sei = mask.sti = mask.ssi = 0;
+            mask.uei = mask.uti = mask.usi = 0;
+            break;
+        default:
+            panic("Unknown privilege mode %d.", prv);
+            break;
+    }
+
+    return std::bitset<NumInterruptTypes>(mask);
+}
+
+Fault
+Interrupts::getInterrupt()
+{
+    assert(checkInterrupts());
+    if (checkNonMaskableInterrupt())
+        return std::make_shared<NonMaskableInterruptFault>();
+    std::bitset<NumInterruptTypes> mask = globalMask();
+    if (((ISA*) tc->getIsaPtr())->rvType() == RV64) {
+        const std::vector<int> interrupt_order {
+            INT_LOCAL_63, INT_LOCAL_62, INT_LOCAL_61, INT_LOCAL_60,
+            INT_LOCAL_59, INT_LOCAL_58, INT_LOCAL_57, INT_LOCAL_56,
+            INT_LOCAL_55, INT_LOCAL_54, INT_LOCAL_53, INT_LOCAL_52,
+            INT_LOCAL_51, INT_LOCAL_50, INT_LOCAL_49, INT_LOCAL_48,
+            INT_LOCAL_47, INT_LOCAL_46, INT_LOCAL_45, INT_LOCAL_44,
+            INT_LOCAL_43, INT_LOCAL_42, INT_LOCAL_41, INT_LOCAL_40,
+            INT_LOCAL_39, INT_LOCAL_38, INT_LOCAL_37, INT_LOCAL_36,
+            INT_LOCAL_35, INT_LOCAL_34, INT_LOCAL_33, INT_LOCAL_32,
+            INT_LOCAL_31, INT_LOCAL_30, INT_LOCAL_29, INT_LOCAL_28,
+            INT_LOCAL_27, INT_LOCAL_26, INT_LOCAL_25, INT_LOCAL_24,
+            INT_LOCAL_23, INT_LOCAL_22, INT_LOCAL_21, INT_LOCAL_20,
+            INT_LOCAL_19, INT_LOCAL_18, INT_LOCAL_17, INT_LOCAL_16,
+            INT_EXT_MACHINE, INT_SOFTWARE_MACHINE, INT_TIMER_MACHINE,
+            INT_EXT_SUPER, INT_SOFTWARE_SUPER, INT_TIMER_SUPER,
+            INT_EXT_USER, INT_SOFTWARE_USER, INT_TIMER_USER
+        };
+        for (const int &id : interrupt_order)
+        if (checkInterrupt(id) && mask[id])
+            return std::make_shared<InterruptFault>(id);
```

Comment:
Please add some curly braces here. It's quite unclear what the intention of this code is, and it's likely to cause future bugs.